File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.js
/**
 * @param {string[]} words
 * @return {boolean}
 */
var makeEqual = function(words) {
    let arr = new Array(26).fill(0);
    for(let word of words) {
        for(let ch of word) {
            arr[ch.charCodeAt(0)-'a'.charCodeAt(0)]++;
        }
    }
    let len = words.length;
    for(let freq of arr) {
        if(freq % len !== 0) return false;
    }
    return true;
};

File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.java
class Solution {
    public boolean makeEqual(String[] words) {
        int[] arr = new int[26];
        int len = words.length;
        for(var word : words) {
            for(char ch : word.toCharArray()) {
                arr[ch-'a']++;
            }
        }
        for(int i=0; i<26; i++) {
            if(arr[i] % len != 0) return false;
        }
        return true;
    }
}